### PR TITLE
Azure Custom Vision Export Provider - Phase 1 - AB#15637

### PR DESCRIPTION
**Pull request by wbreza**
- Export provider boilerplate
- JSON form schema
- External Picker (for dynamic form fields)
- Refactoring on project save
- Creates new ACV project on project settings save

### CODE REVIEW

File: src/providers/export/azureCustomVision.json [hunk 1]
```diff
@@ -1 +1,112 @@
...
+    "type": "object",
+    "title": "Azure Custom Vision",
+    "required": [
+        "assetState",
+        "apiKey"
+    ],
+    "properties": {
+        "assetState": {
+            "type": "string",
+            "title": "Asset State",
```
Comment (wbreza) [inline src/providers/export/azureCustomVision.json]: Will localize strings in next task PR.

File: src/providers/export/azureCustomVision.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+export class AzureCustomVisionProvider extends ExportProvider<IAzureCustomVisionOptions> {
+    constructor(project: IProject, options: IAzureCustomVisionOptions) {
+        super(project, options);
+        Guard.null(options);
+    }
+
+    public export(): Promise<void> {
+        throw new Error("Method not implemented.");
```
Comment (wbreza) [inline src/providers/export/azureCustomVision.ts]: Export method will be implemented in next PR task.

File: src/common/strings.ts [hunk 3]
```diff
@@ -142,35 +141,25 @@ export interface IAppStrings {
...
-        }
-    }
-    return result;
+export function addLocValues(json: any) {
+    return interpolateJson(json, { strings });
 }
 
-export function addLocValues(json: any) {
-    return replaceVariablesInJson(json, getLocValue);
+export function interpolateJson(json: any, params: any) {
```
Comment (wbreza) [inline src/common/strings.ts]: @tbarlow12 - This is the method that replaces the regex parsing and using native template string interpolation.

Comment (tbarlow12) [reply src/common/strings.ts]: Looks great

File: src/providers/export/azureCustomVision.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        const queryString = this.createQueryString(urlParams);
+        const config: AxiosRequestConfig = {
+            headers: {
+                "Training-key": customVisionOptions.apiKey,
+            },
+        };
+
+        // tslint:disable-next-line:max-line-length
+        const url = `https://southcentralus.api.cognitive.microsoft.com/customvision/v2.2/Training/projects?${queryString}`;
```
Comment (tbarlow12) [inline src/providers/export/azureCustomVision.ts]: Is it always `southcentralus`?

**Pull request merged**